src/test/java/com/endava/intership/collections/StudentMapTest.java

```diff
@@ -0,0 +1,189 @@
+            assertAll(
+                    () -> assertThat(testTreeMap).doesNotContainKey(studentsList.get(finalI)),
+                    () -> assertThat(testTreeMap).doesNotContainValue(finalI+1),
+                    () -> assertThat(testTreeMap).hasSize(finalI),
+                    () -> assertThat(testTreeMap.get(studentsList.get(finalI))).isNull()
+            );
+        }
+    }
+
+    @Test
+    void  removeOneTest() {
+        StudentMap<Student, Integer> testTreeMap  = new StudentMap<>();
+        Student expectedStudent1 = new Student("John Doe", LocalDate.of(2000, 5, 12), "");
+        Student expectedStudent2 = new Student("Ne John Doe", LocalDate.of(2000, 5, 12), "");
+
+        testTreeMap.put(expectedStudent1, 1);
+        testTreeMap.remove(expectedStudent1);
+
+        assertAll(
+                () -> assertThat(testTreeMap).isEmpty(),
+                () -> assertThat(testTreeMap).hasSize(0),
+                () -> assertThat(testTreeMap.get(expectedStudent1)).isNull()
+        );
+
+        testTreeMap.put(expectedStudent1, 1);
+        testTreeMap.put(expectedStudent2, 2);
+        testTreeMap.remove(expectedStudent1);
+
+        assertAll(
+                () -> assertThat(testTreeMap).isNotEmpty(),
+                () -> assertThat(testTreeMap).hasSize(1),
+                () -> assertThat(testTreeMap.get(expectedStudent1)).isNull(),
+                () -> assertThat(testTreeMap).containsKey(expectedStudent2),
+                () -> assertThat(testTreeMap.get(expectedStudent2)).isEqualTo(2),
+                () -> assertThat(testTreeMap).containsValue(2)
+        );
+    }
+
+    @Test
+    void sizeTest() {
+        assertEquals(testTreeMap.size(), studentsList.size());
+    }
+
+    @Test
+    void isEmptyTest() {
+        StudentMap<Student, Integer> testTreeMap  = new StudentMap<>();
+        assertTrue(testTreeMap.isEmpty());
+        testTreeMap.put(studentsList.get(0), 0);
+        assertFalse(testTreeMap.isEmpty());
+        testTreeMap.remove(studentsList.get(0));
+        assertTrue(testTreeMap.isEmpty());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentsForTreeMapTests")
+    void containsKeyTest(Student student, int number) {
+        assertTrue(testTreeMap.containsKey(student));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentsForTreeMapTests")
+    void getTest(Student student, int number) {
+        assertEquals(testTreeMap.get(student), number);
+    }
+
+    @Test
+    void putAllTest() {
+        HashMap<Student, Integer> testHashMap = new HashMap<>();
+        for (int i = 0; i < studentsList.size(); i++) {
+            testHashMap.put(studentsList.get(i), i+1);
+        }
+        StudentMap<Student, Integer> testTreeMap  = new StudentMap<>();
+        testTreeMap.putAll(testHashMap);
+        assertEquals(testTreeMap.size(), testHashMap.size());
+    }
+
+    @Test
+    void clearTest() {
+        StudentMap<Student, Integer> testTreeMap  = new StudentMap<>();
```

Comment:
Why do you repeat here step from beforeAll method?

Reply:
I fixed it